packages/dds/merge-tree/src/test/obliterate.rangeExpansion.spec.ts

```diff
@@ -0,0 +1,383 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { ReconnectTestHelper } from "./reconnectHelper.js";
+import { TestClient } from "./testClient.js";
+
+describe("obliterate", () => {
+	let client: TestClient;
+	let events: number[];
+
+	beforeEach(() => {
+		client = new TestClient({
+			mergeTreeEnableObliterate: true,
+		});
+		events = [];
+		client.startOrUpdateCollaboration("local");
+		for (const char of "hello world") {
+			client.applyMsg(
+				client.makeOpMessage(
+					client.insertTextLocal(client.getLength(), char),
+					client.getCurrentSeq() + 1,
+				),
+			);
+		}
+		assert.equal(client.getText(), "hello world");
+	});
+
+	it("obliterate adjacent insert", () => {
```

Comment:
All of these tests seem to have a similar structure, which I think would be good to factor out.

```typescript
function itCorrectlyObliterates({
  title: string,
  action: (helper: ReconnectTestHelper) => void,
  expectedText: string,
  expectedEventCount: number
}) {
  return it(title, () => {
    const helper = new ReconnectTestHelper();
    helper.clients.A.on("delta", (opArgs, deltaArgs) => {
      events.push(deltaArgs.operation);
    });
    action(helper);
    helper.processAllOps();
    assert.equal(helper.clients.A.getText(), expectedText);
    assert.equal(helper.clients.C.getText(), expectedText);
    assert.equal(events.length, expectedEventCount, `events: ${events.join(", ")}`);
  });
}
```

Reply:
In fact, doing it this way we could get rid of the `beforeEach` entirely. `itCorrectlyObliterates` can do all of the required setup locally.